pruner/light/pruner.go

```diff
@@ -3,15 +3,30 @@ package light
 import (
 	"context"
 
+	"github.com/ipfs/boxo/blockservice"
+
 	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share/ipld"
 )
 
-type Pruner struct{}
+type Pruner struct {
+	bserv blockservice.BlockService
+}
 
-func NewPruner() *Pruner {
-	return &Pruner{}
+func NewPruner(bserv blockservice.BlockService) *Pruner {
+	return &Pruner{bserv: bserv}
 }
 
-func (p *Pruner) Prune(context.Context, *header.ExtendedHeader) error {
+func (p *Pruner) Prune(ctx context.Context, h *header.ExtendedHeader) error {
+	var roots [][]byte
+	roots = append(roots, h.DAH.RowRoots...)
+	roots = append(roots, h.DAH.ColumnRoots...)
+	for _, root := range roots {
+		cid := ipld.MustCidFromNamespacedSha256(root)
+		if err := ipld.DeleteNode(ctx, p.bserv, cid); err != nil {
+			return err
+		}
+	}
```

Comment:
That only happens on failure though

Reply:
>Amd making the pruner responsible to delete that data if sampling never succeeds is very debatable. Because that data was never available and once its outside of the window, we like never care about data unavalability and just proceed. Feels a bit wrong.
Also, making the pruner aware of implementation details of availability feels a bit wrong

Isn't purpose of availability window to remove responsibility of node to or guarantee any info about availability outside of the window? If we leave this info in datastore it will essentially leak, because it will never be removed or used in any way, polluting the store. Once the pruner passed the specific height sampling info will never be accessed.  

I don't like the idea of making pruner aware of availability details either, but it is rather puzzle for implementation how to do it. One of the way to accomplish it is to add method to `Availability` interface. It can also serve as Store related interface that was discussed in https://github.com/celestiaorg/celestia-node/pull/3388#discussion_r1596667868

Reply:
Ok, adding cleanup